src/test/regress/expected/multi_name_lengths.out

```diff
@@ -125,12 +125,69 @@ SELECT "Constraint", "Definition" FROM table_checks WHERE relid='public.name_len
 ALTER TABLE name_lengths RENAME TO name_len_12345678901234567890123456789012345678901234567890;
-ERROR:  shard name name_len_12345678901234567890123456789012345678_fcd8ab6f_xxxxx exceeds 63 characters
+ERROR:  Shard name (name_len_12345678901234567890123456789012345678_fcd8ab6f_xxxxx) for table (name_len_12345678901234567890123456789012345678901234567890) is too long and could lead to deadlocks when executed in a transaction block after a parallel query
+HINT:  Try re-running the transaction with "SET LOCAL citus.multi_shard_modify_mode TO 'sequential';"
+ROLLBACK;
+-- The same operation will work when sequential mode is set
+BEGIN;
+SET LOCAL citus.multi_shard_modify_mode TO 'sequential';
+ALTER TABLE name_lengths rename col1 to new_column_name;
+ALTER TABLE name_lengths RENAME TO name_len_12345678901234567890123456789012345678901234567890;
+DEBUG:  the name of the shard (name_len_12345678901234567890123456789012345678_fcd8ab6f_xxxxx) for relation (name_len_12345678901234567890123456789012345678901234567890) is too long, switching to sequential and local execution mode to prevent self deadlocks
+ROLLBACK;
+RESET client_min_messages;
+-- test long partitioned table renames
+CREATE TABLE partition_lengths
+(
+    tenant_id integer NOT NULL,
+    timeperiod timestamp without time zone NOT NULL
+) PARTITION BY RANGE (timeperiod);
+SELECT create_distributed_table('partition_lengths', 'tenant_id');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE TABLE partition_lengths_p2020_09_28 PARTITION OF partition_lengths FOR VALUES FROM ('2020-09-28 00:00:00') TO ('2020-09-29 00:00:00');
+-- verify that we can rename partitioned tables and partitions to too-long names
+ALTER TABLE partition_lengths RENAME TO partition_lengths_12345678901234567890123456789012345678901234567890;
+NOTICE:  identifier "partition_lengths_12345678901234567890123456789012345678901234567890" will be truncated to "partition_lengths_123456789012345678901234567890123456789012345"
+-- verify that we can rename partitioned tables and partitions with too-long names
+ALTER TABLE partition_lengths_12345678901234567890123456789012345678901234567890 RENAME TO partition_lengths;
+NOTICE:  identifier "partition_lengths_12345678901234567890123456789012345678901234567890" will be truncated to "partition_lengths_123456789012345678901234567890123456789012345"
+-- Placeholders for unsupported operations
+\set VERBOSITY TERSE
+-- renaming distributed table constraints are not supported
 ALTER TABLE name_lengths RENAME CONSTRAINT unique_12345678901234567890123456789012345678901234567890 TO unique2_12345678901234567890123456789012345678901234567890;
 ERROR:  renaming constraints belonging to distributed tables is currently unsupported
+-- renaming distributed table partitions are not supported
+ALTER TABLE partition_lengths_p2020_09_28 RENAME TO partition_lengths_p2020_09_28_12345678901234567890123456789012345678901234567890;
```

Comment:
Should we create these tables with replication factor == 1, in that case it should be supported?

Reply:
I removed the test cases where replication factor > 1